storage/block/types.go

```diff
@@ -185,6 +186,29 @@ type DatabaseBlock interface {
 
 	// Close closes the block.
 	Close()
+
+	// SetOwner sets the owner of the block
+	SetOwner(Owner)
+
+	// Owner returns the owner of the block
+	Owner() Owner
+
+	// Private methods because only the Wired List itself should use them.
+	next() DatabaseBlock
+	setNext(block DatabaseBlock)
+	prev() DatabaseBlock
+	setPrev(block DatabaseBlock)
+	nextPrevUpdatedAtUnixNano() int64
+	setNextPrevUpdatedAtUnixNano(value int64)
+	wiredListEntry() wiredListEntry
+}
+
+// Owner is implemented by an "owner" of a block that should
+// be notified when a block is evicted from the wired list (and thus unavailable
+// for reads.)
+type Owner interface {
```

Comment:
Would `Listener` by a more apt name?

Reply:
I thought Owner was a little more clear cause it conveyed the fact that the block isnt owned by the wired list...maybe EvictionListener?

Reply:
Could just call this: